docs/GettingStartedDocs/Contributors/SignUsingEngines.md

```diff
@@ -0,0 +1,114 @@
+# Signing a Build With OpenSSL Engines
+
+OpenSSL allows extensions called "engines" to perform the generation of keys, signing and encryption.
+Engines are useful. They can be used to allow hardware offload of encryption duties, or improved 
+encapsulation of secrets. Use of engines is required in some build systems as a way to avoid the
+storage of keys in plain text .pem files.
+
+oesign has been equipped with the ability to specify engines in the enclave signing process. In order
+to use the engine, the command line syntax has been extended.
+```
+oesign sign --enclave-image ENCLAVE_IMAGE --config-file CONFIG_FILE --engine engine-NAME -load-path engine-LOAD-PATH-key KEY-ID
+
+```
+
+## Engines 
+
+The concept of an engine was added to OpenSSL starting with 0.9.6. There are built in engine implementations
+implementations for
+- Microsoft CryptoAPI
+- VIA Padlock
+- nCipher CHIL
+and a api for controlling dynamically bound engines which are bound from shared libraries at runtime via either program logic
+or configuration files. 
+
+The openssl commnd line utility can show available engines via the "engine" command.
+```
+# openssl engine
+(rdrand) Intel RDRAND engine
+(dynamic) Dynamic engine loading support
+
+```
+Of particular interest is the "dynamic" engine, which allow dynamic plugins to perform encryption duties. 
+These objects allow a plugin mechanism for implementations of cryptographic algorithms and can be statically linked in the
+OpenSSL library at compile-time or dynamically loaded at run-time in and out of the running application 
+with low overhead from external binary shared objects implementing the engine API using special built-in engine
+called “dynamic”. 
+
+Dynamic engines are particularly interesting due to the flexibility they provide: 
```

Comment:
Here has dynamic "engines" plural but line 32 uses dynamic engine singular with dynamic "plugins" plural.  Should this say "Dynamic plugins are..."?

Reply:
There is an engine named "dynamic" which connect to different provided dynamic engines.
Plugin is an adjetive to introduce the idea. Dynamic engine(s) are the term of art.